cardano-node/src/Cardano/Node/Tracers.hs

```diff
@@ -132,6 +80,104 @@ data Tracers peer blk = Tracers {
 
 -- | Smart constructor of 'NodeTraces'.
 --
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => Transformable Text IO (WithTip blk (ChainDB.TraceEvent blk)) where
+    trTransformer = trStructured -- structure required, will call 'toObject'
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => ToObject (WithTip blk (ChainDB.TraceEvent blk)) where
+    toObject MinimalVerbosity _ = emptyObject -- no output
+    toObject verb (WithTip tip ev) =
+        mkObject [ "tip" .= showTip tip
+                 , "TraceEvent" .= toObject verb ev
+                 ]
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => ToObject (ChainDB.TraceEvent blk) where
+    toObject MinimalVerbosity _ = emptyObject -- no output
+    toObject _ (ChainDB.TraceAddBlockEvent ev) =
```

Comment:
A sketch of less verbose (pun intended) alternative approach:
```haskell
    toObject verbosity ev = case verbosity of
        MinimalVerbosity -> emptyObject -- no output
        MaximalVerbosity -> mkObj (constrName ev) (show ev) -- NEW: useful for debugging
        NormalVerbosity  -> case ev of
          ChainDB.TraceAddBlockEvent ev' -> mkObj "TraceAddBlockEvent" $ \case ev' of
            ChainDB.StoreButDontChange   pt ->
              "Ignoring block: " <> condense pt
            ChainDB.TryAddToCurrentChain pt ->
              "Block fits onto the current chain: " <> condense pt
        ..
      where
        mkObj kind event = mkObject
          [ "kind"  .= String kind
          , "event" .= String $ pack event
          ]
```

Reply:
👍 Thanks! I will revise.
now it looks like code. before, it was just a verbatim copy of the existing case-split.